src/App.js

```diff
@@ -10,10 +11,36 @@ const tabs = [
 ];
 /* eslint-enable */
 
-const App = () => (
-  <div className="App">
-    <h1>Selected tab is&nbsp;Tab 1</h1>
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    selectedTab: tabs[0],
+  }
+
+  changeSelectedTab = (tabId) => {
+    this.setState(() => {
+      const newSelectedTab = tabs.find(tab => tab.id === tabId);
+
+      return {
+        selectedTab: newSelectedTab,
+      };
```

Comment:
![image](https://user-images.githubusercontent.com/35385361/120029306-9659d480-bffe-11eb-8a53-87fe2aabdcf7.png)
if user clicks on the tab that already selected setState shouldn't be called